server/api/src/rsDataObjChksum.cpp

```diff
@@ -34,34 +38,22 @@ namespace {
         }
         const rodsLong_t size_in_vault = fileStatOut->st_size;
         free(fileStatOut);
-        genQueryInp_t genQueryInp;
-        memset(&genQueryInp, 0, sizeof(genQueryInp));
-        addInxIval(&genQueryInp.selectInp, COL_DATA_SIZE, 1);
-        addInxVal(&genQueryInp.sqlCondInp, COL_D_DATA_PATH, (boost::format(" = '%s'") % dataObjInfo->filePath).str().c_str());
-        genQueryInp.maxRows = 1;
-        genQueryOut_t *genQueryOut = nullptr;
-        const int status_rsGenQuery = rsGenQuery(rsComm, &genQueryInp, &genQueryOut);
-        clearGenQueryInp(&genQueryInp);
-        if (status_rsGenQuery < 0) {
-            if (status_rsGenQuery == CAT_NO_ROWS_FOUND) {
-                freeGenQueryOut(&genQueryOut);
+        std::optional<rodsLong_t> size_in_database {};
+        std::string select_and_condition {
+            (boost::format("select DATA_SIZE where DATA_PATH = '%s'") % dataObjInfo->filePath).str()
+        };
+        for (auto&& row : irods::query{rsComm, select_and_condition}) {
+            try {
+                size_in_database = boost::lexical_cast<rodsLong_t>(row[0]);
+            } catch (const boost::bad_lexical_cast&) {
+                rodsLog(LOG_ERROR, "_rsFileChksum: lexical_cast of [%s] for [%s] [%s] [%s] failed", row[0].c_str(), dataObjInfo->filePath, dataObjInfo->rescHier, dataObjInfo->objPath);
```

Comment:
On the `master` branch, use the new logger instead of `rodsLog`. This line could be:
```c++
using log = irods::experimental::log; // Just after #includes.
                                      // Could also be another name if "log" is already used.

// Here is how you'd use the new logger.
// You can use a different category if you like. "api" is not required to be used here, but just used
// as an example. See "irods_logger.hpp" for the full list of categories.
log::api::error("_rsFileChksum: lexical_cast of [{}] for [{}] [{}] [{}] failed",
                row[0],
                dataObjInfo->filePath,
                dataObjInfo->rescHier,
                dataObjInfo->objPath);
```